addons/mail/static/tests/helpers/mock_server.js

```diff
@@ -1032,16 +1224,71 @@ MockServer.include({
             };
         });
     },
+    /**
+     * Simulates `_message_compute_author` on `mail.thread`.
+     *
+     * @private
+     * @param {string} model
+     * @param {integer[]} ids
+     * @param {Object} [context={}]
+     * @return {integer}
+     */
+    _MockMailThread_MessageComputeAuthor(model, ids, author_id, email_from, context = {}) {
+        if (author_id === undefined) {
+            // For simplicity, partner is not guessed from email_from here, but
+            // that is would be the first step on the server.
+            let user_id;
+            if ('mockedUserId' in context) {
+                // can be falsy to simulate not being logged in
+                user_id = context.mockedUserId
+                    ? context.mockedUserId
+                    : this.publicUserId;
+            } else {
+                user_id = this.currentUserId;
+            }
+            const user = this._getRecords(
+                'res.users',
+                [['id', '=', user_id]],
+                { active_test: false },
+            )[0];
+            const author = this._getRecords(
+                'res.partner',
+                [['id', '=', user.partner_id]],
+                { active_test: false },
+            )[0];
+            author_id = author.id;
+            email_from = `${author.display_name} <${author.email}>`;
+        }
+
+        if (email_from === undefined) {
+            if (author_id) {
+                const author = this._getRecords(
+                    'res.partner',
+                    [['id', '=', author_id]],
+                    { active_test: false },
+                )[0];
+                email_from = `${author.display_name} <${author.email}>`;
+            }
+        }
+
+        if (!email_from) {
+            throw Error("Unable to log message");
```

Comment:
Would clarify what is wrong (something to do with lacking recipient info)
Also technically not just a "log" but ok in that it broadens to the general concept of "history logs" (and it matches wording of error in server code).

Reply:
It's missing author info actually not recipient info, I'll add that to the error message.